bigchaindb/models.py

```diff
@@ -376,6 +392,28 @@ def decouple_assets(self):
 
         return (assets, block_dict)
 
+    def decouple_metadata(self, block_dict=None):
+        """
+        Extracts the metadata from transactions in the block.
+
+        Returns:
+            tuple: (metadatas, block) with the metadatas being a list of dict/null and
+            the block being the dict of the block with no metadata in any transaction.
+        """
+        if block_dict is None:
+            block_dict = deepcopy(self.to_dict())
+
+        metadatas = []
+        for transaction in block_dict['block']['transactions']:
+            metadata = transaction.pop('metadata')
+            if isinstance(metadata, dict):
+                metadata.update({'id': transaction['id']})
```

Comment:
We should probably nest the metadata under a new key or else if the metadata contains a key named `id` this will override it. We can probably have a data model similar to the asset. E.g.:
```python
{
     'id': '<transaction id>'
    'metadata': '<the metadata dict>'
}
```

Reply:
agreed.